dspace-api/src/main/java/org/dspace/identifier/ezid/MetadataFixupTask.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.identifier.ezid;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.content.DSpaceObject;
+import org.dspace.content.Item;
+import org.dspace.content.MetadataSchema;
+import org.dspace.content.MetadataValue;
+import org.dspace.content.factory.ContentServiceFactory;
+import org.dspace.content.service.DSpaceObjectService;
+import org.dspace.core.Context;
+import org.dspace.curate.AbstractCurationTask;
+import org.dspace.curate.Curator;
+import org.dspace.curate.Mutative;
+import org.dspace.identifier.DOI;
+import org.dspace.identifier.EZIDIdentifierProvider;
+import org.dspace.identifier.IdentifierException;
+import org.dspace.identifier.IdentifierProvider;
+import org.dspace.identifier.factory.IdentifierServiceFactory;
+import org.dspace.identifier.service.DOIService;
+import org.dspace.services.ConfigurationService;
+import org.dspace.utils.DSpace;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A curation task to fix incorrect use of {@code dc.identifier} for holding
+ * DSpace-generated EZID DOIs.  That field is reserved for user-provided
+ * identifiers.  Earlier versions of EZID support incorrectly used that field
+ * to hold DOIs that it generates.
+ *
+ * @author mwood
+ */
+@Mutative
+public class MetadataFixupTask
+        extends AbstractCurationTask
```

Comment:
As noted in IRC, it might be worth considering reworking this into a Flyway (Java or SQL) migration.  A curation task would "work", but a Flyway migration is more automated.  Plus, it seems like a "run once" sort of thing, which may fit better under a Flyway migration instead of a curation task.